### ISSUE TITLE
Ivy-posframe does not truncate lines (cursor goes out of frame when typing long lines)

### ISSUE THREAD

**Issue by ianyepan** [tumashu/ivy-posframe#123]
As titled, ivy-posframe doesn't seem to support "truncate lines" and when I'm typing a long file name, the cursor can disappear from the right edge of the screen (but I'm still able to continue typing). Here's a screenshot to explain myself:

![image](https://user-images.githubusercontent.com/36686900/154323872-d0b6a3dd-4be9-4321-a2ab-24c9689c92ce.png)

## Here's how ivy normally handles it.

Ivy would firstly try to put the long line on the next line, if the space is still not enough, it'll scroll right as the user types, keeping the **cursor always visible at the right edge of the frame**, and pushing text on the left side out of the screen. Basically how you'd type a long URL in a browser textbox -- the start of the URL might not be visible, but the cursor should always be visible. Let me know if I'm explaining this clearly enough. Cheers!

**Comment by tumashu** [comment]
Yes, because "truncate lines" looks very ugly,  posframe support it with lines-truncate argument, but ivy-posframe do not use it.

**Comment by ianyepan** [comment]
I see, then is it possible to keep the cursor visible at all times (at the right edge of the frame while pushing left edge's content out of the frame, like normal textbox design). I've tried to describe this behavior in my last paragraph above -- let me know what you think!
I'm actually surprised that no one brought this up (am I missing something?). Surely people have had to find-file in a deeply nested folder structure and hence having a long file name that goes off the screen.

**Comment by jrasband** [comment]
This is also an issue for me. It's possible to see the full file name if you set `ivy-truncate-lines` to `nil`, but it looks very ugly. I think that the suggested behavior from @ianyepan would be ideal.
